### ISSUE TITLE
[Bug] Android 13 (and later versions): Unable to locate Mansions of Madness / Unable to locate Road to Legend

### ISSUE THREAD

**Issue by ParkerLewis31884** [NPBruce/valkyrie#1602]
### Describe the bug
Valkyrie loads fine, but is "Unable to locate Mansions of Madness" even though the official Android MoM app (1.9.0) is normally installed.

EDIT : MoM just updated again to 1.9.1. No change (same issue, same behavior).

### Steps to reproduce the behavior:
1. Load Valkyrie
2. There is no step 2

### Expected behavior
Valkyrie should find the official MoM app.

### Valkyrie Version
2.5.8

### Smartphone
 - Device: Pixel 4a
 - OS: Android 13 build TQ1A.221202.011

### Additional context
MoM (1.9.0, then 1.9.1) was configured in french in the app. I tried setting it back in english (and relaunching MoM, waiting to reach main screen, then relaunching Valkyrie). Did not make any difference. Not sure if / how that could be relevant but mentioning it anyway.

Otherwise, I see no possible relevant context. Phone is stock Pixel4a with up to date Android 13. MoM installed normally from the Play Store and up to date. No shenanigans. Installed the app exactly as requested, and allowed all authorizations when prompted (and manually to get Valkyrie to reach its main screen).

If I click on "Install via Google play", it opens the MoM page in the Play Store as expected. The Play Store as expected recognizes the app as installed already.

**Comment by ivansengir** [comment]
Exactly same problem here with a Samsung A52s Andorid 13

**Comment by Quantumrunner** [comment]
Android 13 is currently not supported because there is no way to access the MoM data folder. Wiki has been updated:

https://github.com/NPBruce/valkyrie/wiki/Troubleshooting#blackscreen-on-startup-android

**Comment by ParkerLewis31884** [comment]
> Android 13 is currently not supported because there is no way to access the MoM data folder. Wiki has been updated:
> 
> https://github.com/NPBruce/valkyrie/wiki/Troubleshooting#blackscreen-on-startup-android

This is a section about blackscreen on startup which does not match the issue. Linked section of the wiki says

> For all later versions (Android 13 and above) the app will only show a black screen on startup. There is currently no workaround for this.

But that is not what is happening here. Applying the exact workaround described for Android 11/12 (ie giving Valkyrie the "install app" privilege), the Valkyrie app does load fine and reaches / displays its main menu fine. What it doesn't do as expected though is being able to locate the MoM app.

**Comment by Quantumrunner** [comment]
Yes you are right the issue behaves differently but it is caused by the same root cause:

Android is preventing the app from getting access to other apps data. For Android 11 and 12 [a workaround](https://github.com/NPBruce/valkyrie/pull/1576/files) could be used but it seems that in Android 13 the issue has reappeared because Google stopped this workaround.

I will update the wiki page to make clearer that this new issue will appear in Android 13.

**Comment by ParkerLewis31884** [comment]
Ah, ok. Thanks for the clarification.

I will be asking a question I know is pretty naive, but I'm interested in the answer / getting why this couldn't work. If one were to manually copy the MoM app data in another custom (and thus unprotected ?) location, would it be possible for Valkyrie to then just look there (possibly requesting the user to provide the path to look for) ?

**Comment by Quantumrunner** [comment]
> manually copy the MoM app data in another custom location (possibly [this way](https://www.droidwin.com/how-to-access-android-13-data-and-obb-folder-transfer-files-in-them/)), would it be possible for Valkyrie to then just look there (possibly requesting the user to provide the path to look for) ?

Yes this would be the easiest solution. Letting the user pick the location himself is not easy but the could ask him to put the data in the download folder of his device (which does need any extra permissions). But even this simple change is not possible with quite some code changes.

**Comment by Quantumrunner** [comment]
Closed. Issue has been fixed in Release https://github.com/NPBruce/valkyrie/releases/tag/release%2F2.6%2F2.6.0a.

**Comment by Quantumrunner** [comment]
Closed. Issue has been fixed in Release https://github.com/NPBruce/valkyrie/releases/tag/release%2F2.6%2F2.6.0a.

**Comment by seinsinnes** [comment]
My experience with new versions of Android 14 suggests that the loop holes which allowed the reading of other app's data have been closed.

Unless someone knows more tricks I don't see how we'll be able to support valkyrie importing data from the official apps moving forward.
